cli/azd/cmd/actions/middleware.go

```diff
@@ -13,15 +13,19 @@ type NextFn func(ctx context.Context) (*ActionResult, error)
 type MiddlewareFn func(ctx context.Context, buildOptions *ActionOptions, next NextFn) (*ActionResult, error)
 
 // Executes the middleware chain for the specified action
-func RunWithMiddleware(ctx context.Context, buildOptions *ActionOptions, action Action, chain []MiddlewareFn) (*ActionResult, error) {
-	chainLength := len(chain)
+func RunWithMiddleware(
```

Comment:
This is the new function that executes the full middleware chain.  This is would be invoked automatically during the `root` command and manually for all sub-actions within a composite command.